orbit/pkg/update/nudge.go

```diff
@@ -223,6 +238,17 @@ func (n *NudgeConfigFetcher) launch() error {
 			}
 
 			if err := fn(nudge.DirPath, fmt.Sprintf("file://%s", cfgFile)); err != nil {
+				if exitErr, ok := err.(*exec.ExitError); ok {
+					launchErr := &nudgeLaunchErr{
+						err:       err,
+						exitCode:  exitErr.ExitCode(),
+						detail:    string(exitErr.Stderr),
+						cfgFile:   cfgFile,
+						timestamp: time.Now(),
+					}
+					n.launchErr = launchErr
+					return fmt.Errorf("opening Nudge with config %q: %w", cfgFile, launchErr)
```

Comment:
I think the errors where we are most concerned with disabling Nudge are the ones that cause the application to steal focus. Non-exit errors (i.e. errors happening outside the Nudge process) shouldn't be causing the focus to shift. 

The upside of disabling for non-exit errors would be potentially avoiding the overhead of checking the file hash each loop. But we'd need to think through the range of errors because some might potentially resolve with time (e.g., missing JSON config). So far, the product discussion has been narrowly focused on the focus-stealing problem.

Reply:
> The upside of disabling for non-exit errors would be potentially avoiding the overhead of checking the file hash each loop. But we'd need to think through the range of errors because some might potentially resolve with time (e.g., missing JSON config). So far, the product discussion has been narrowly focused on the focus-stealing problem.

I'm not worried about the the overhead, but I'm worried about errors that might cause similar problems. Agreed that we would need to think a bit about it, but for example a missing JSON config won't throw an error (Nudge will use the default config).